fix default netty thread in docs as 0 and minor markdown syntax issue

rishabh1212 opened the pull request:
## Description
- In the docs [configuration.md](https://github.com/pytorch/serve/blob/master/docs/configuration.md), it is mentioned that default netty thread number is the number of available logical processors
- According to [ConfigManager.java](https://github.com/pytorch/serve/blob/bacdf0a626fc24d89f332c413a2039f7c0cf9953/frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java#L316), the Netty threads set are 0, and nowhere else they are set as ```Runtime.getRuntime().availableProcessors();```.
- Only the default number of workers are set as the number of logical processors.

Fixes https://github.com/pytorch/serve/issues/782

## Type of change

Please delete options that are not relevant.

- [ ] This change requires a documentation update

## Feature/Issue validation/testing

- 

## Checklist:

- [ ] Have you added tests that prove your fix is effective or that this feature works?
- [ ] New and existing unit tests pass locally with these changes?
- [ ] Has code been commented, particularly in hard-to-understand areas?
- [ ] Have you made corresponding changes to the documentation?

rishabh1212 commented:
@harshbafna @dhaniram-kshirsagar I made a change to documentation. Can you please review?

rishabh1212 commented:
@harshbafna @dhaniram-kshirsagar @maaquib All checks are passed. Is there anything I should do next? I am sorry this is my first time making a PR to open source!

rishabh1212 commented:
@harshbafna @dhaniram-kshirsagar Can you please clarify on whether https://github.com/pytorch/serve/issues/782 is actually an issue or just my misunderstanding about Netty threads?

dhaniram-kshirsagar commented:
> @harshbafna @dhaniram-kshirsagar Can you please clarify on whether #782 is actually an issue or just my misunderstanding about Netty threads?

@rishabh1212 The modification you have made are incorrect. However, it seems like our doc is indeed incorrect a bit. Here is netty code snippet which does trick of using available number of processor to jvm when we specify number of threads as 0 from torchserve configmanager. 
```java
 DEFAULT_EVENT_LOOP_THREADS = Math.max(1, SystemPropertyUtil.getInt(
                "io.netty.eventLoopThreads", NettyRuntime.availableProcessors() * 2));

Please correct those doc lines according to above snippet.

rishabh1212 commented:
> @rishabh1212 The modification you have made are incorrect. However, it seems like our doc is indeed incorrect a bit. Here is netty code snippet which does trick of using available number of processor to jvm when we specify number of threads as 0 from torchserve configmanager.
> 
> ```java
>  DEFAULT_EVENT_LOOP_THREADS = Math.max(1, SystemPropertyUtil.getInt(
>                 "io.netty.eventLoopThreads", NettyRuntime.availableProcessors() * 2));
> 
> Please correct those doc lines according to above snippet.
> ```

@dhaniram-kshirsagar thanks for your comment. I fixed the docs in PR and also updated description of PR. Please review

harshbafna reviewed (changes requested):
@rishabh1212: This looks good. Could you also update the default values in the `dumpConfiguration` method of ConfigManager? As it prints the default value as 0 at TorchServe startup and might create confusion.

rishabh1212 commented:
> @rishabh1212: This looks good. Could you also update the default values in the `dumpConfiguration` method of ConfigManager? As it prints the default value as 0 at TorchServe startup and might create confusion.

@harshbafna I made the changes, please review

harshbafna commented on frontend/server/src/main/java/org/pytorch/serve/util/ConfigManager.java:
```diff
@@ -572,9 +578,9 @@ public String dumpConfigurations() {
                 + getCanonicalPath(System.getProperty("METRICS_LOCATION"))
                 + "\nNetty threads: "
-                + getNettyThreads()
+                + (getNettyThreads() == 0 ? DEFAULT_EVENT_LOOP_THREADS : getNettyThreads())
```
Can't we simply return the default value set in Netty's MultithreadEventLoopGroup class?

rishabh1212 replied:
@harshbafna I did not want to change the way Netty calculate the value. I could have returned `2*runtime.availableProcessors()`.  For example, in future if they change the value returned the function `NettyRuntime.availableProcessors`, then the value printed by us will not be consistent with they function they use to calculate default.

harshbafna commented:
@rishabh1212: The build is failing because of a formatting issue. You can use the following commands:

```
cd frontend
./gradlew fJ
```

rishabh1212 commented:
> @rishabh1212: The build is failing because of a formatting issue. You can use the following commands:
> 
> ```
> cd frontend
> ./gradlew fJ
> ```

Thanks, I was trying to fix it.

rishabh1212 commented:
@harshbafna, thanks! I was able to fix it. Please check https://github.com/pytorch/serve/pull/873#discussion_r544047443 and review

rishabh1212 commented:
@msaroufim I have made requested fixes with some [explanation](https://github.com/pytorch/serve/pull/873#issuecomment-746422289). I think its ready for review
